### PR TITLE
Deprecate nxos_portchannel and add nxos_linkagg DI module

### LINKED ISSUE #31887: [DI_NXOS] Deprecate nxos_portchannel and ADD nxos_linkagg (non-DI to DI)
Deprecate non-DI nxos_portchannel -> Add DI module nxos_linkagg
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

modules/network/nxos/nxos_portchannel
##### ANSIBLE VERSION

```
2.5
```

### PR DESCRIPTION
**Pull request by trishnaguha**
##### SUMMARY

fixes https://github.com/ansible/ansible/issues/31887
- Deprecate nxos_portchannel
- Add nxos_linkagg DI module https://github.com/ansible/ansible/issues/31303
- Integration test 

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

modules/network/nxos/nxos_portchannel
modules/network/nxos/nxos_linkagg
test/integration/targets/nxos_linkagg
##### ANSIBLE VERSION

```
devel
```

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_linkagg.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = """
+---
+module: nxos_portchannel
```
Comment (gundalow) [inline lib/ansible/modules/network/nxos/nxos_linkagg.py]: `module: nxos_linkagg`
Is what's causing the
`lib/ansible/modules/network/nxos/nxos_linkagg.py:0:0: E305 DOCUMENTATION.module: not a valid value for dictionary value @ data['module']. Got 'nxos_portchannel'`

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Ansible Changes By Release
 * `stat` and `win_stat` have deprecated `get_md5` and the `md5` return value
   and these options will be removed in Ansible 2.9. `get_md5: no` will still be
   allowed in 2.9 but will finally be removed 2 versions after that.
+* nxos_portchannel module is deprecated in Ansible 2.5. Use nxos_linkagg module instead.
```
Comment (gundalow) [inline CHANGELOG.md]: Please also add to `docs/docsite/rst/porting_guide_2.5.rst`

**Review by gundalow — CHANGES_REQUESTED** [review]
Do we have unit tests/integration tests for nxos_linkagg?

File: CHANGELOG.md (hunk 2 above)
Comment (trishnaguha) [reply CHANGELOG.md]: Added.

**Review by Qalthos — CHANGES_REQUESTED** [review]
No tests for nxos_linkagg? Also, I would keep the portchannel tests around until we remove the code entirely

**Comment by trishnaguha** [comment]
@Qalthos Integration test WIP. Added nxos_portchannel unit test back.

**Review by gundalow — APPROVED** [review]

**Comment by trishnaguha** [comment]
@Qalthos Please review now. All the tests are passing along with nxos_portchannel and newly added nxos_linkagg.

**Review by gundalow — APPROVED** [review]

**Review by Qalthos — APPROVED** [review]

**Pull request merged**